### ISSUE TITLE
Compatibility with 2D videos

### ISSUE THREAD

**Issue by jclement46** [gkv311/sview#65]
3D has the difficulty that you have to use a separate viewer for 3D than for 2D.  To make it more accessible it would be nice to have one viewer for both formats.  Since sView supports both tags and suffixes to identify file types a suffix -2D could be used to label 2D videos and slides.  In addition the extension .JPS is used for 3D, so .JPG could be used by default for 2D and JPS for 3D.

This should not be too difficult as it would just be a matter of making the single image be duplicated for R & L display before the display section.  While the PC user might have to don glasses, the Android user with a glasses free display could seamlessly view both 2D and 3D intermixed.  An added selectable feature of having 2D behind the frame might be nice, but just being able to intermix would be OK.

Of course it would be best if the manufacturers would supply seamless software, but at present there is little hope for that.

**Comment by gkv311** [comment]
I don't quite understand the problem description - it looks exactly how sView currently works when stereoscopic format is set to "Source" (which is default):
- Videos and image without tags are displayed in mono.
- Files with special tags in name (any image of video) / flags (MKV) / metadata (WMV) / SEI (MP4) / streams (MPO) / chunks (PNG / JPEG) / file extension (.jps, .pns).

So what is proposed to be changed / improved?

**Comment by jclement46** [comment]
If I have a standard 2D movie, sView assumes it is 3D and tries to play it in 3D SBS.  If I have a 2D picture it also tries to display it in 3D SBS.  This is the way virtually all 3D players work.  So to display 2D I have to use a different program.  What I want is a single program that will display 2D and 3D intermixed.

 

To do this it would be reasonable to have it selectable for JPS to denote 3D, and JPG for 2D, as this is already fairly standard.  Videos are more problematic because the tags are often not included, so a suffix could be used to denote 2D, or it could default to 2D in the absence of the appropriate tag or suffix (-LRQ,…).  Since 3D is practically never encoded in BMP, TIF,…. And mainly in JPG or JPS those other formats could default to 2D.

 

Another option would be just to have a 2D/3D button.  This could be part of the R/L button as a third option for 2D. A fourth selection could be just 2D display of one side of a 3D movie.  However, this does not replace automatic switching between 2D and 3D display modes.

 

As I pointed out this should not require cutting into the 3D display part, but rather the front end where the R/L images are formatted for 3D display.  As a programmer I would assume that the decoding and display are separate routines.

 

This would make sView a universal slide/movie viewer rather than just a 3D viewer.  Some included software in 3D phones have very limited universal viewers which do not handle all formats and options.  I have both 2D and 3D pictures, so having a display that would switch automatically would be a great enhancement.

 

Regards,

John

**Comment by gkv311** [comment]
> If I have a standard 2D movie, sView assumes it is 3D and tries to play it in 3D SBS.
> If I have a 2D picture it also tries to display it in 3D SBS. 

No, sView shouldn't guess 2D image or video to be SBS if stereoscopic format is set to "Source", and not manually selected format. When user specifies input format manually - it is applied than to all files.

.jps is already treated as SBS and .jpg as 2d. The same for video having -sbs and other suffixes in name or with proper tags inside.

**Comment by jclement46** [comment]
This is was not obvious to me.  I checked it out and it works, but what is the suffix for a 2D video?  I just looked at your site and can not find any info about the suffixes.

 

Thanks for the info,

John

**Comment by gkv311** [comment]
>  I just looked at your site and can not find any info about the suffixes.

Take a look onto this page:
http://www.sview.ru/en/help/input/

> but what is the suffix for a 2D video?

There is no need for an extra suffix for 2D video - it just shouldn't have any tag used for 3D format.
But technically sView also checks for **-2d** suffix for  this purpose.

**Comment by jclement46** [comment]
OK, I got the photo formats to work correctly.  But 2D videos are still being displayed in 3D, which of course is a mess.  I tried appending “-2d” to them, and it didn’t work.   I checked other output formats and they didn’t work either.

 

The -2d suffix is not included in the page you referred me to.  I would suggest also that the document should mention that the explicit tags such as -LRQ should be at the end of the file name,  assuming it is like TriDef which also required that.

 

Regards,

John

**Comment by gkv311** [comment]
> But 2D videos are still being displayed in 3D, which of course is a mess.

Could you please share a screenshot with such use case with sView settings visible (Stereoformat menu)?
